File: App.Models/Explanation.cs
using App.Core.Models;

namespace App.Models
{
    public class Explanation : IAuditableEntity
    {
        public int Id { get; set; }
        public int ComplaintId { get; set; }
        public int SubmittedBy { get; set; }
        public string ExplanationText { get; set; } = string.Empty;
        public DateTime SubmittedAt { get; set; }
        public bool IsComplainant { get; set; }
        public DateTime CreatedOn { get; set; }
        public DateTime ModifiedOn { get; set; }

        // Navigation properties
        public virtual Complaint Complaint { get; set; } = null!;
        public virtual User Submitter { get; set; } = null!;
    }
}

File: App.client/projects/proctor-app/scripts/models/explanation.ts
import { User } from './user';
import { Complaint } from './complaint';

export class Explanation {
  id: number;
  complaintId: number;
  submittedBy: number;
  explanationText: string;
  submittedAt: string;
  isComplainant: boolean;
  createdOn: string;
  modifiedOn: string;
  complaint?: Complaint | null;
  submitter?: User | null;

  constructor() {
    this.id = 0;
    this.complaintId = this.submittedBy = 0;
    this.explanationText = '';
    this.submittedAt = new Date().toISOString();
    this.isComplainant = false;
    this.createdOn = this.modifiedOn = '';
  }
}

